### PR TITLE
Use func instead of proc where appropriate

### PR DESCRIPTION
**Pull request by vabresto**
Hey, thanks for this great library!

One of my favourite Nim features is the side effects tracking the compiler does, so I made these changes for my own code and wanted to share them back.

I didn't make any changes other than `proc` -> `func`. All the tests pass.

Also, VS Code warns that the various functions cannot raise `Defect` (this is new as of 1.4.0 it seems) - https://nim-lang.org/blog/2020/10/16/version-140-released.html - I thought about just removing them but that's probably better in another PR (not sure if there's any safety value in keeping them around if the compiler doesn't track it anymore). Let me know if you'd like me to include that in this PR as a general clean up.

Thanks!

### CODE REVIEW

**Comment by GULPF** [comment]
Thanks for the contribution, seems like a good change.

> Let me know if you'd like me to include that in this PR as a general clean up.

That would be great :)

**Comment by vabresto** [comment]
Wanted to follow up on this so it's not left hanging - I can't seem to be able to trigger CI runs, but the devel build seems to have failed for an unrelated reason. Is there anything waiting on me or that I can do to help?

File: nimquery.nim [hunk 1]
```diff
@@ -912,7 +912,7 @@ proc parseHtmlQuery*(queryString: string,
                 else: doAssert(false) # can't happen
 
             of CombinatorKinds:
-                parts.add initQueryPart(demands, lexer.current.kind.ord.Combinator)
+                parts.add initQueryPart(demands, lexer.current.kind.Combinator)
```
Comment (GULPF) [inline nimquery.nim:915]: This change is why the devel build is failing - Nim won't allow direct enum to enum conversions in the future. Revert this and it should work

**Review by GULPF — CHANGES_REQUESTED** [review]

**Comment by SolitudeSF** [comment]
compilation now fails with `Error: 'attrComparerString' can have side effects`

**Pull request merged**